### ISSUE TITLE
`prove` and `verify` for evm pipeline

### ISSUE THREAD

**Issue by alexander-camuto** [zkonduit/ezkl#104]
Currently the evm pipeline (activated with `--features evm` )  is only enabled for `fullprove` whereby the proof and verifying key (and other parameters) are keep in memory and never serialized to disk ! 

We need to similarly enable separate `prove` and `verify` commands specific to evm verification ! When calling `prove` this would involve serializing the proof generated by 
```rust
let proof = gen_kzg_proof::<
                        _,
                        _,
                        EvmTranscript<G1Affine, _, _, _>,
                        EvmTranscript<G1Affine, _, _, _>,
                    >(
                        &params, &pk, agg_circuit.clone(), agg_circuit.instances()
                    )?;
```
the aggregate circuit instances generated from: 

```rust
let instances = agg_circuit.instances();
``` 
 and the verification smart contract deployment code: 

```rust 
let deployment_code = gen_aggregation_evm_verifier(
                        &params,
                        pk.get_vk(),
                        AggregationCircuit::num_instance(),
                        AggregationCircuit::accumulator_indices(),
                    )?;
```

Calling `verify`would then read these serialized values from disk and call: 

```rust
evm_verify(deployment_code, agg_circuit.instances(), proof)?;
```

In summary this involves the following tasks: 

- [x] separate `prove` and `verify` pathways in `src/execute.rs` for when `--features evm` is activated (for a template for how this is done, see `fullprove` in the same file)
- [x] serializing the aggregate circuit instances and saving to a file
- [x] serializing the aggregate circuit proof and saving to a file
- [x] serializing the verification smart contract deployment code and saving to file
- [x] `verify` should read all 3 of the above files and call `evm_verify`
- [x] add integration tests which test the new pipeline
